From ccc3695932f132db656b5b9721561a3e3b7bfc14 Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@redhat.com>
Date: Wed, 23 Jul 2014 10:49:56 -0400
Subject: [virt] virtio_ring: adapt to notify() returning bool

Message-id: <1406112611-9862-9-git-send-email-brueckner@redhat.com>
Patchwork-id: 86320
O-Subject: [RHEL7.1 PATCH 08/23] [virt] virtio_ring: adapt to notify() returning bool
Bugzilla: 1088554
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Description
-----------
commit 2342d6a6512ce5a3d2433bf77e6580e738cfd709
Author: Heinz Graalfs <graalfs@linux.vnet.ibm.com>
Commit: Rusty Russell <rusty@rustcorp.com.au>

    virtio_ring: adapt to notify() returning bool

    Correct if statement to check for bool returned by notify()
    (introduced in 5b1bf7cb673a).

    Signed-off-by: Heinz Graalfs <graalfs@linux.vnet.ibm.com>
    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

Bugzilla
--------
BZ 1088554
https://bugzilla.redhat.com/show_bug.cgi?id=1088554

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version v3.13-rc1
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=2342d6a6512ce5a3d2433bf77e6580e738cfd709

Brew
----
https://brewweb.devel.redhat.com/taskinfo?taskID=7725759

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/virtio/virtio_ring.c b/drivers/virtio/virtio_ring.c
index a687469..a224d64 100644
--- a/drivers/virtio/virtio_ring.c
+++ b/drivers/virtio/virtio_ring.c
@@ -473,7 +473,7 @@ bool virtqueue_notify(struct virtqueue *_vq)
   return false;
 
  /* Prod other side to tell it about changes. */
- if (vq->notify(_vq) < 0) {
+ if (!vq->notify(_vq)) {
   vq->broken = true;
   return false;
  }
-- 
1.7.1